### PR TITLE
Add deployment GitHub action.

### PR DESCRIPTION
**Pull request by bpblanken**
Just a scaffold here... 

I'm more or less blocked on permissioning:
Screenshot 2023-12-07 at 5 31 11 PM

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.workflow_run.head_branch }}
+
+      - name: authenticate to google cloud
+        id: "auth"
+        uses: google-github-actions/auth@v0
```
Comment (bpblanken) [inline .github/workflows/deploy.yaml:25]: Some details on how this works [here](https://cloud.google.com/blog/products/identity-security/enabling-keyless-authentication-from-github-actions)

**Comment by theferrit32** [comment]
Added `IAM Workload Identity Pool Admin` to `clingendevs` group

**Comment by bpblanken** [comment]
I'm taking a shot at the permissions with terraform: https://github.com/clingen-data-model/architecture/pull/599/files

**Comment by bpblanken** [comment]
Merging so that we can start testing!

**Pull request merged**